ngraph/core/include/ngraph/op/if.hpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+
+#include "ngraph/function.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/util/multi_subgraph_base.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v8
+        {
+            /// \brief  If operation.
+            class NGRAPH_API If : public util::MultiSubGraphOp
+            {
+            public:
+                enum BodyIndexes
+                {
+                    then_body_index = 0,
+                    else_body_index = 1
+                };
+
+                NGRAPH_RTTI_DECLARATION;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                /// \brief     Constructs If with condition
+                ///
+                /// \param     execution_condition   condition node.
+                If(const Output<Node>& execution_condition);
+                If();
```

Comment:
@itikhono, do we need both constructors?

Reply:
usually we use If() constructor in IE IR reader.
I guess both are necessary
@evolosen please check it